libdrgn/dwarf_index.c

```diff
@@ -199,23 +199,44 @@ static inline size_t hash_pair_to_shard(struct hash_pair hp)
 	 */
 	return ((hp.first >>
 		 (8 * sizeof(size_t) - 8 - DRGN_DWARF_INDEX_SHARD_BITS)) &
-		(((size_t)1 << DRGN_DWARF_INDEX_SHARD_BITS) - 1));
+		(((size_t)DRGN_DWARF_INDEX_NUM_SHARDS) - 1));
 }
 
 static void
 drgn_dwarf_index_namespace_init(struct drgn_dwarf_index_namespace *ns,
 				struct drgn_dwarf_index *dindex)
 {
-	array_for_each(shard, ns->shards) {
-		omp_init_lock(&shard->lock);
-		drgn_dwarf_index_die_map_init(&shard->map);
-		drgn_dwarf_index_die_vector_init(&shard->dies);
-	}
+	ns->shards = NULL;
 	ns->dindex = dindex;
 	drgn_dwarf_index_pending_die_vector_init(&ns->pending_dies);
 	ns->saved_err = NULL;
 }
 
+static struct drgn_error *
```

Comment:
This can only fail if it can't allocate memory, so can you please make it return a `bool` (`true` for success, `false` if no memory)?